src/Testcontainers/Builders/HttpWaitRequest.cs

```diff
@@ -0,0 +1,137 @@
+namespace DotNet.Testcontainers.Builders
+{
+  using System;
+  using System.Collections.Generic;
+  using System.Diagnostics.CodeAnalysis;
+  using System.Net.Http;
+  using JetBrains.Annotations;
+
+  /// <summary>
+  /// A http wait request.
+  /// </summary>
+  [PublicAPI]
+  public class HttpWaitRequest
+  {
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HttpWaitRequest" /> class.
+    /// </summary>
+    /// <param name="port">The private container port.</param>
+    /// <param name="statusCodes">The expected status codes.</param>
+    /// <param name="path">The absolute path of the request uri.</param>
+    /// <param name="method">The http method.</param>
+    /// <param name="readTimeout">the http connection read timeout.</param>
+    private HttpWaitRequest(int port, ISet<int> statusCodes, string path, HttpMethod method, TimeSpan readTimeout)
+    {
+      this.Port = port;
+      this.StatusCodes = statusCodes;
+      this.Path = path;
+      this.Method = method;
+      this.ReadTimeout = readTimeout;
+    }
+
+    public int Port { get; }
+
+    public string Path { get; }
+
+    public HttpMethod Method { get; }
+
+    public ISet<int> StatusCodes { get; }
+
+    public TimeSpan ReadTimeout { get; }
+
+    /// <summary>
+    /// Returns a new instance of wait request builder for a given port
+    /// </summary>
+    /// <param name="port">The private container port.</param>
+    /// <returns>Instance of <see cref="HttpWaitRequest.Builder" />.</returns>
+    [PublicAPI]
+    public static Builder ForPort(int port)
+    {
+      return new Builder(port);
+    }
+
+    /// <summary>
+    /// A fluent wait request builder.
+    /// </summary>
+    [SuppressMessage("ReSharper", "ParameterHidesMember", Justification = "Fluent builder")]
+    [PublicAPI]
+    public class Builder
+    {
+      private const string DefaultPath = "/";
+
+      private readonly int port;
```

Comment:
I think you can use the [UriBuilder](https://docs.microsoft.com/en-us/dotnet/api/system.uribuilder?view=net-6.0) here and store the values in the builder field.